src/entities/Employee.java

```diff
@@ -0,0 +1,37 @@
+package entities;
+
+import java.util.Objects;
+
+public class Employee extends User{
+    private Long code;
```

Comment:
Hmm have you thought about auditory and tracing? I would suggest not to consider this field in the equals and the hashcode since these field could exist multiple times with the same code given the different statuses